contracts/token/ERC20/ERC20.sol

```diff
@@ -153,9 +153,11 @@ contract ERC20 is Context, IERC20, IERC20Metadata {
         uint256 amount
     ) public virtual override returns (bool) {
         uint256 currentAllowance = _allowances[sender][_msgSender()];
-        require(currentAllowance >= amount, "ERC20: transfer amount exceeds allowance");
-        unchecked {
-            _approve(sender, _msgSender(), currentAllowance - amount);
+        if (currentAllowance != 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff) {
```

Comment:
```suggestion
        if (currentAllowance != type(uint256).max) {
```

Reply:
Why not 
`if (currentAllowance !=type(uint256).max) {`

IMO that would be more explicit

Reply:
See my comment above, `.max` requires 0.8.8.

Reply:
0.8.8 introduced max for enums. For value types like uint256 is available in 0.8.0 ... and we are already using it in many places.

Reply:
Oh nice, I read the readme too quickly.